### PR TITLE
Project menu bar fix

### PR DESCRIPTION
**Pull request by dregad**
Fixes the following issues

- [0026887](https://mantisbt.org/bugs/view.php?id=26887): Project Menu Bar does not indent subprojects properly
- [0026888](https://mantisbt.org/bugs/view.php?id=26888): Refactor printing of project selection menus

### CODE REVIEW

File: core/layout_api.php [hunk 1]
```diff
@@ -633,6 +635,7 @@ function layout_navbar_subproject_option_list( $p_parent_id, $p_project_id = nul
 	array_push( $p_parents, $p_parent_id );
 	$t_user_id = auth_get_current_user_id();
 	$t_project_ids = user_get_accessible_subprojects( $t_user_id, $p_parent_id );
+	$t_indent = str_repeat( '&nbsp;',4 );
```
Comment (vboctor) [inline core/layout_api.php:638]: Fix spacing after the comma.

File: core/project_api.php [hunk 2]
```diff
@@ -936,3 +936,39 @@ function project_format_id( $p_project_id ) {
...
+ */
+function project_link_for_menu( $p_project_id, $p_active = false, $p_class = '', array $p_parents = array(), $p_indent = '' ) {
+	if( $p_parents ) {
+		$t_full_id = implode( ";", $p_parents ) . ';' . $p_project_id;
+		$t_indent = str_repeat( $p_indent, count( $p_parents ) ) . '&nbsp;';
+	} else {
+		$t_full_id = $p_project_id;
+		$t_indent = '';
+	}
+	$t_url = helper_mantis_url( 'set_project.php?project_id=' . $t_full_id );
```
Comment (vboctor) [inline core/project_api.php:965]: Consider adding a blank line before this one and deleting the blank line after this one.

File: core/layout_api.php (hunk 1 above)
Comment (dregad) [reply core/layout_api.php]: Fixed

File: core/project_api.php (hunk 2 above)
Comment (dregad) [reply core/project_api.php]: Fixed

**Comment by dregad** [comment]
> I would prefer not having more view code in project_api and moving that out to print_api, html_api, or layout_api as appropriate.

I did think about this before putting the new function in project_api... IMO it does not belong

- in print_api, because the new function does not actually *print* anything, it returns a formatted string
- in layout_api because it has nothing to do with page layout

html_api would be the "least bad" choice, but I still think it makes sense to have it in project_api.

Function can still be moved later on if necessary, so this is not blocking this PR's merge.

**Pull request merged**